doc/Makefile

```diff
@@ -114,12 +114,7 @@ html-build:
 	$(SPHINXBUILD) -WT --keep-going $(VERSIONWARNING) -b html $(ALLSPHINXOPTS) build/html $(FILES)
 
 convert-notebooks:
-	mkdir -p source/_contents
-	for file in source/tutorial/stats/*.md; do \
-		basename=$$(basename $$file .md); \
-		output_name="source/_contents/$${basename}.ipynb"; \
-		$(PYTHON) -m jupytext --output "$$output_name"  $$file; \
-	done
+	$(PYTHON) convert_notebooks.py
```

Comment:
Let me check if your code caches the output of `jupytext`. Basically the `make` code in `main` branch always calls `jupytext` irrespective of whether the input `md` are unchanged. That is redundant. My Python script checks whether any modifications in an `md` file and calls `jupytext` on it accordingly. I use simple timestamp comparisons (one can get last modification timestamp in the input file (`.md`) and creation timestamp of output file (`.ipynb`). If modification happened after creation of output then `jupytext` needs to be called otherwise not.

Reply:
I just checked the `make` code in https://github.com/scipy/scipy/pull/21433#discussion_r1763210692 does the same thing as `main` i.e., always calls `jupytext` irrespective of whether any modification didn't happen in any of the `md` file. The Python script calls `jupytext` only when needed i.e., if any `md` file was modified.

Reply:
Oops, I did not test carefully enough :sweat_smile: I accidentally typed ``/*.md`` instead of ``/%.md``. The following correction produces the desired behavior:
```Makefile
source/_contents/%.ipynb : source/tutorial/stats/%.md
```

Reply:
@DietBru my `make` foo isn't super fluent, but is that actually the same? The Python script does timestamp-based caching, which is what we need. Ideally the tool we invoke (`jupytext` for `.md` -> `.ipynb`) caches, and if that's not available then we implement a custom cache. It seems the latter is done in a `.py` script here.

Reply:
That is what is happening. From the [make manual]:

> A target is out of date if it does not exist or if it is older than any of the prerequisites (by comparison of last-modification times). 

E.g., the following example show that touch one file causes only this file to be rebuilt:
```shell

$  make convert-notebooks   # everything is up to date
make: Nothing to be done for 'convert-notebooks'.
$ touch source/tutorial/stats/hypothesis_bartlett.md   # change timestamp of one file
$ make convert-notebooks   # only  one file is rebuilt
[jupytext] Reading source/tutorial/stats/hypothesis_bartlett.md in format md
[jupytext] Writing source/_contents/hypothesis_bartlett.ipynb (destination file replaced [use --update to preserve cell outputs and ids])
$ 
```




[make manual]: https://www.gnu.org/software/make/manual/html_node/Rule-Syntax.html

Reply:
When I execute, `python dev.py doc -j8` after doing clean build of SciPy, it doesn't cache output of `jupytext`. I am using the following change in `Makefile`,

```zsh
# List of all .md files, which are to be converted into .ipynb files:
mdFILES := $(notdir $(wildcard source/tutorial/stats/*.md))

convert-notebooks: $(mdFILES:%.md=source/_contents/%.ipynb)

source/_contents/%.ipynb : source/tutorial/stats/%.md
	@mkdir -p source/_contents/  # create directory if needed
	@$(PYTHON) -m jupytext --output $@ $<
```

Reply:
Weird :thinking: 
If the behavior of the Makefile is not easy to predict, I guess it is easier to stick to the Python script. My idea was that having the build rules all in one place would make live simpler. Sorry for the noise.

Reply:
Which version of `make` are you using?

Reply:
```shell-session
$ make --version
GNU Make 4.3
Built for x86_64-pc-linux-gnu
```

I would be surprised if it is a version problem, though.

Reply:
I tried with the following `make` and it didn't make difference. It still doesn't cache.

```zsh
GNU Make 4.3
Built for arm64-apple-darwin20.0.0
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```